applications/SwimmingDEMApplication/custom_elements/qsvms_dem_coupled.cpp

```diff
@@ -0,0 +1,859 @@
+        }
+    }
+
+template <class TElementData>
+void QSVMSDEMCoupled<TElementData>::CalculateLeftHandSide(MatrixType& rLeftHandSideMatrix,
+                                                            ProcessInfo& rCurrentProcessInfo)
+{
+    // Resize and intialize output
+    QSVMS<TElementData>::CalculateLeftHandSide(rLeftHandSideMatrix, rCurrentProcessInfo);
+    if (rCurrentProcessInfo[FRACTIONAL_STEP] != 1) {
+        noalias(rLeftHandSideMatrix) = ZeroMatrix(LocalSize, LocalSize);
+        CalculateLaplacianMassMatrix(rLeftHandSideMatrix, rCurrentProcessInfo);
+    }
+}
+
+template<class TElementData>
+void QSVMSDEMCoupled<TElementData>::CalculateRightHandSide(VectorType& rRightHandSideVector,
+                                                            ProcessInfo& rCurrentProcessInfo)
+    {
+        TElementData data;
+        data.Initialize(*this, rCurrentProcessInfo);
+        // Calculate this element's geometric parameters
+        double Weight = data.Weight;
+        array_1d<double, 3> convective_velocity;
+
+        // Calculate this element's fluid properties
+        QSVMS<TElementData>::CalculateRightHandSide(rRightHandSideVector, rCurrentProcessInfo);
+
+        if (rCurrentProcessInfo[FRACTIONAL_STEP] == 1) {
+
+            //this->AddMomentumRHS(rRightHandSideVector, Weight, data);
+            const double& DeltaTime = rCurrentProcessInfo[DELTA_TIME];
+            static const double arr[] = {1.0,-1.0};
+            std::vector<double> SchemeWeights (arr, arr + sizeof(arr) / sizeof(arr[0]));
+
+            this->AddMassRHS(rRightHandSideVector, data.Density, data.N, Weight, SchemeWeights, DeltaTime, data);
+        }
+        else {
+            const unsigned int LocalSize = Dim * NumNodes;
+
+            // Check sizes and initialize
+            if (rRightHandSideVector.size() != LocalSize)
+                rRightHandSideVector.resize(LocalSize, false);
+
+            noalias(rRightHandSideVector) = ZeroVector(LocalSize);
+            this->AddRHSLaplacian(rRightHandSideVector, data.DN_DX, Weight);
+        }
+        if (data.UseOSS == 1.0)
+
+            this->GetAdvectiveVel(convective_velocity, data.N);
+
+            double tau_one, tau_two;
+            this->CalculateTau(data, convective_velocity, tau_one, tau_two);
+
+            this->AddProjectionToRHS(rRightHandSideVector, convective_velocity, data, tau_one, tau_two, Weight, rCurrentProcessInfo[DELTA_TIME]);
+    }
+
+template<class TElementData>
+void QSVMSDEMCoupled<TElementData>::AddMassStabilization(TElementData& rData,
+                                                        MatrixType &rMassMatrix)
+    {
+
+        const double density = this->GetAtCoordinate(rData.Density, rData.N);
+
+        double tau_one;
+        double tau_two;
+        array_1d<double, 3> convective_velocity=
+        this->GetAtCoordinate(rData.Velocity, rData.N) -
+        this->GetAtCoordinate(rData.MeshVelocity, rData.N);
```

Comment:
This is hard to read. Please use indentations if you want to do this operation in multiple lines.